tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -161,13 +166,37 @@ def resources(self, query=None):
                     local_session(self.session_factory).get_default_project())
                 return []
             raise
-        self._cache.save(key, resources)
-        return self.filter_resources(resources)
 
     def augment(self, resources):
         return resources
 
 
+class ChildResourceManager(QueryResourceManager):
+    def _fetch_resources(self, query):
+        if not query:
+            query = {}
+
+        resources = []
+
+        parent_description = self.resource_type.parent
+        parent_resource_type, parent_field, arg_name = parent_description
+        injection_name = 'c7n:{}'.format(parent_resource_type)
+        resource_manager = self.get_resource_manager(parent_resource_type)
+
+        for parent in resource_manager.resources():
+            arg_value = jmespath.search(parent_field, parent)
+            query.update({arg_name: arg_value})
+
+            children = super(ChildResourceManager, self)._fetch_resources(query)
+
+            for child in children:
+                child[injection_name] = parent
```

Comment:
the other issue here with storing the full value is that now policies are not isomorphic wrt to their runtime execution, ie. a serverless policy on a resource, won't have this attribute per the example sql db resource.

Reply:
Among `get` method I've had to override `ChildResourceManager#get_resource`. I cannot retrieve parent using its corresponding `get` method since I don't have a client.
Take a look please.